modules/repository-s3/src/internalClusterTest/java/org/elasticsearch/repositories/s3/S3RepositoryThirdPartyTests.java

```diff
@@ -243,15 +244,15 @@ public void testReadFromPositionLargerThanBlobLength() {
 
         var exception = expectThrows(UncategorizedExecutionException.class, () -> readBlob(repository, blobName, position, length));
         assertThat(exception.getCause(), instanceOf(ExecutionException.class));
-        assertThat(exception.getCause().getCause(), instanceOf(IOException.class));
+        assertThat(exception.getCause().getCause(), instanceOf(RequestedRangeNotSatisfiedException.class));
         assertThat(
             exception.getCause().getCause().getMessage(),
             containsString(
-                "Requested range [start="
+                "Requested range [position="
                     + position
-                    + ", end="
-                    + (position + length - 1L)
-                    + ", currentOffset=0] cannot be satisfied for blob object ["
+                    + ", length="
+                    + (length - 1L)
```

Comment:
```suggestion
                    + (length)
```

Does this need the -1? I think the `end` which was previously needed it, but not the length.